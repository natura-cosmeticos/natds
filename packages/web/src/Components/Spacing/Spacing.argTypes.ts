import { ArgType } from '@storybook/addons'
import { ISpacingProps } from './Spacing.props'
import spacingPropArgType from '../../../.storybook/argTypes/spacingPropArgType'

type ArgTypesKey = keyof ISpacingProps | string;

export const spacingArgTypes : Record<ArgTypesKey, ArgType> = {
  margin: spacingPropArgType,
  padding: spacingPropArgType,
  marginBottom: spacingPropArgType,
  marginLeft: spacingPropArgType,
  marginRight: spacingPropArgType,
  marginTop: spacingPropArgType,
  marginX: spacingPropArgType,
  marginY: spacingPropArgType,
  paddingBottom: spacingPropArgType,
  paddingLeft: spacingPropArgType,
  paddingRight: spacingPropArgType,
  paddingTop: spacingPropArgType,
  paddingX: spacingPropArgType,
  paddingY: spacingPropArgType
}

export const spacingAliasArgTypes : Record<ArgTypesKey, ArgType> = {
  // eslint-disable-next-line id-length
  m: spacingPropArgType,
  mb: spacingPropArgType,
  ml: spacingPropArgType,
  mr: spacingPropArgType,
  mt: spacingPropArgType,
  mx: spacingPropArgType,
  my: spacingPropArgType,
  // eslint-disable-next-line id-length
  p: spacingPropArgType,
  pb: spacingPropArgType,
  pl: spacingPropArgType,
  pr: spacingPropArgType,
  pt: spacingPropArgType,
  px: spacingPropArgType,
  py: spacingPropArgType
}

export const argTypes : Record<ArgTypesKey, ArgType> = {
  ...spacingArgTypes,
  children: {},
  component: {},
  ...spacingAliasArgTypes
}

export default argTypes
