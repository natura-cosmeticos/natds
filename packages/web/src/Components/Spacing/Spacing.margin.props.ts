import { SizePropValue } from '../../hooks/useSizeStyleProp'

export type SpacingMarginProps = {

  /**
   * Alias for `margin` property
   *
   * @type SizePropValue
   */
  m?: SizePropValue,

  /**
   * Defines margin
   *
   * @type SizePropValue
   */
  margin?: SizePropValue,

  /**
   * Defines bottom margin
   *
   * @type SizePropValue
   */
  marginBottom?: SizePropValue,

  /**
   * Defines left margin
   *
   * @type SizePropValue
   */
  marginLeft?: SizePropValue,

  /**
   * Defines right margin
   *
   * @type SizePropValue
   */
  marginRight?: SizePropValue,

  /**
   * Defines top margin
   *
   * @type SizePropValue
   */
  marginTop?: SizePropValue,

  /**
   * Defines left and right margins
   *
   * @type SizePropValue
   */
  marginX?: SizePropValue,

  /**
   * Defines bottom and top margins
   *
   * @type SizePropValue
   */
  marginY?: SizePropValue,

  /**
   * Alias for `margin-bottom` property
   *
   * @type SizePropValue
   */
  mb?: SizePropValue,

  /**
   * Alias for `margin-left` property
   *
   * @type SizePropValue
   */
  ml?: SizePropValue,

  /**
   * Alias for `margin-right` property
   *
   * @type SizePropValue
   */
  mr?: SizePropValue,

  /**
   * Alias for `margin-top` property
   *
   * @type SizePropValue
   */
  mt?: SizePropValue,

  /**
   * Alias for `marginX` property
   *
   * @type SizePropValue
   */
  mx?: SizePropValue,

  /**
   * Alias for `marginY` property
   *
   * @type SizePropValue
   */
  my?: SizePropValue,
};
