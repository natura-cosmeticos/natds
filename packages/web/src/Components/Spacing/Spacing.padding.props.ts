import { SizePropValue } from '../../hooks/useSizeStyleProp'

export type SpacingPaddingProps = {

  /**
   * Alias for `padding` property
   *
   * @type SizePropValue
   */
  p?: SizePropValue,

  /**
   * Defines padding
   *
   * @type SizePropValue
   */
  padding?: SizePropValue,

  /**
   * Defines bottom padding
   *
   * @type SizePropValue
   */
  paddingBottom?: SizePropValue,

  /**
   * Defines left padding
   *
   * @type SizePropValue
   */
  paddingLeft?: SizePropValue,

  /**
   * Defines right padding
   *
   * @type SizePropValue
   */
  paddingRight?: SizePropValue,

  /**
   * Defines top padding
   *
   * @type SizePropValue
   */
  paddingTop?: SizePropValue,

  /**
   * Defines left and right paddings
   *
   * @type SizePropValue
   */
  paddingX?: SizePropValue,

  /**
   * Defines bottom and top paddings
   *
   * @type SizePropValue
   */
  paddingY?: SizePropValue,

  /**
   * Alias for `padding-bottom` property
   *
   * @type SizePropValue
   */
  pb?: SizePropValue,

  /**
   * Alias for `padding-left` property
   *
   * @type SizePropValue
   */
  pl?: SizePropValue,

  /**
   * Alias for `padding-right` property
   *
   * @type SizePropValue
   */
  pr?: SizePropValue,

  /**
   * Alias for `padding-top` property
   *
   * @type SizePropValue
   */
  pt?: SizePropValue,

  /**
   * Alias for `paddingX` property
   *
   * @type SizePropValue
   */
  px?: SizePropValue,

  /**
   * Alias for `paddingY` property
   *
   * @type SizePropValue
   */
  py?: SizePropValue,
}
