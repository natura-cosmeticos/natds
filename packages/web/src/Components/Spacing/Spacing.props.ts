import * as React from 'react'
import { SpacingMarginProps } from './Spacing.margin.props'
import { SpacingPaddingProps } from './Spacing.padding.props'

export type SpacingMarginCssPropKey = 'marginBottom' | 'marginLeft' | 'marginRight' | 'marginTop';
export type SpacingPaddingCssPropKey = 'paddingBottom' | 'paddingLeft' | 'paddingRight' | 'paddingTop';
export type SpacingStylePropKey = SpacingMarginCssPropKey | SpacingPaddingCssPropKey
export type SpacingMarginAndPaddingProps = SpacingMarginProps & SpacingPaddingProps;

export interface ISpacingProps extends SpacingMarginAndPaddingProps {

  /**
   * Used to render elements inside the Spacing.
   * This can be an element, or just a string.
   *
   * @optional
   * @type React.ReactNode
   */
  children?: React.ReactNode,

  /**
   * Specifies one or more class names for the element.
   *
   * Will be forwarded to the HTML element as the `class` attribute.
   *
   * @optional
   */
  className?: string,

  /**
   * The component used for the root node.
   * Either a string to use a HTML element of a component.
   */
  component?: React.ElementType,

  display?: 'block' | 'flex' | 'inline' | 'inline-block' | 'inline-flex' | Record<string, string>,

  /**
   * Specifies an inline style for an element.
   *
   * The `style` property will override any style set globally.
   * Please avoid using this property on production, prefer `className`.
   *
   * Will be forwarded to the HTML element as the `style` attribute.
   *
   * @optional
   * @type React.CSSProperties
   */
  style?: React.CSSProperties,

}
