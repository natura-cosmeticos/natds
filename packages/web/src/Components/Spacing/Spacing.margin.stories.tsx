import { tokens } from '@naturacosmeticos/natds-styles'
import { Meta, Story } from '@storybook/react'
import { ISpacingProps } from './Spacing.props'
import Template from './Template'
import meta from './Spacing.stories.meta'

export default {
  ...meta,
  title: 'Utilities/Spacing/Margin'
} as Meta

const { sizes } = tokens

export const VeryHugeMargin : Story<ISpacingProps> = Template.bind({})
VeryHugeMargin.args = {
  margin: 'veryHuge'
}
VeryHugeMargin.storyName = `Very Huge Margin (${sizes.veryHuge}px)`

export const HugeMargin : Story<ISpacingProps> = Template.bind({})
HugeMargin.args = { margin: 'huge' }
HugeMargin.storyName = `Huge Margin (${sizes.huge}px)`

export const HugeTopMargin : Story<ISpacingProps> = Template.bind({})
HugeTopMargin.args = { marginTop: 'huge' }
HugeTopMargin.storyName = `Huge Top Margin (${sizes.huge}px)`

export const LargeXHorizontalMargin : Story<ISpacingProps> = Template.bind({})
LargeXHorizontalMargin.args = { marginX: 'largeX' }
LargeXHorizontalMargin.storyName = `LargeX Horizontal Margin (${sizes.largeX}px)`

export const LargeRightMargin : Story<ISpacingProps> = Template.bind({})
LargeRightMargin.args = { marginRight: 'large' }
LargeRightMargin.storyName = `Large Right Margin (${sizes.large}px)`

export const SemiXMarginY : Story<ISpacingProps> = Template.bind({})
SemiXMarginY.args = { marginY: 'semiX' }
SemiXMarginY.storyName = `SemiX Vertical Margin (${sizes.semiX}px)`

export const SemiBottomMargin : Story<ISpacingProps> = Template.bind({})
SemiBottomMargin.args = { marginBottom: 'semi' }
SemiBottomMargin.storyName = `Semi Bottom Margin (${sizes.semi}px)`

export const StandardVerticalMargin : Story<ISpacingProps> = Template.bind({})
StandardVerticalMargin.args = { marginY: 'standard' }
StandardVerticalMargin.storyName = `Standard Vertical Margin (${sizes.standard}px)`

export const StandardLeftMargin : Story<ISpacingProps> = Template.bind({})
StandardLeftMargin.args = { marginLeft: 'standard' }
StandardLeftMargin.storyName = `Standard Left Margin (${sizes.standard}px)`

export const SmallHorizontalMargin : Story<ISpacingProps> = Template.bind({})
SmallHorizontalMargin.args = { marginX: 'small' }
SmallHorizontalMargin.storyName = `Small Horizontal Margin (${sizes.small}px)`

export const SmallTopMargin : Story<ISpacingProps> = Template.bind({})
SmallTopMargin.args = { marginTop: 'small' }
SmallTopMargin.storyName = `Small Top Margin (${sizes.small}px)`

export const SmallRightMargin : Story<ISpacingProps> = Template.bind({})
SmallRightMargin.args = { marginRight: 'small' }
SmallRightMargin.storyName = `Small Right Margin (${sizes.small}px)`

export const TinyMargin : Story<ISpacingProps> = Template.bind({})
TinyMargin.args = { margin: 'tiny' }
TinyMargin.storyName = `Tiny Margin (${sizes.tiny}px)`

export const TinyVerticalMargin : Story<ISpacingProps> = Template.bind({})
TinyVerticalMargin.args = { marginY: 'tiny' }
TinyVerticalMargin.storyName = `Tiny Vertical Margin (${sizes.tiny}px)`

export const TinyRightMargin : Story<ISpacingProps> = Template.bind({})
TinyRightMargin.args = { marginRight: 'tiny' }
TinyRightMargin.storyName = `Tiny Right Margin (${sizes.tiny}px)`

export const MicroBottomMargin: Story<ISpacingProps> = Template.bind({})
MicroBottomMargin.args = { marginBottom: 'micro' }
MicroBottomMargin.storyName = `Micro Bottom Margin (${sizes.micro})`

export const MicroRightMargin : Story<ISpacingProps> = Template.bind({})
MicroRightMargin.args = { marginRight: 'micro' }
MicroRightMargin.storyName = `Micro Right Margin (${sizes.micro})`

export const NoMargin : Story<ISpacingProps> = Template.bind({})
NoMargin.args = { margin: 'none' }
NoMargin.storyName = `No Margin (${sizes.none}px)`
