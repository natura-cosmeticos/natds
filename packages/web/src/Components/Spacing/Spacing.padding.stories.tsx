import { Meta, Story } from '@storybook/react'
import { tokens } from '@naturacosmeticos/natds-styles'
import { ISpacingProps } from './Spacing.props'
import Template from './Template'
import meta from './Spacing.stories.meta'

export default {
  ...meta,
  title: 'Utilities/Spacing/Padding'
} as Meta

const {
  huge, large, micro, none, semi, small, standard, tiny
} = tokens.sizes

export const HugePadding : Story<ISpacingProps> = Template.bind({})
HugePadding.args = { padding: 'huge' }
HugePadding.storyName = `Huge padding (${huge}px)`

export const LargePadding : Story<ISpacingProps> = Template.bind({})
LargePadding.args = { padding: 'large' }
LargePadding.storyName = `Large padding (${large}px)`

export const SemiPadding : Story<ISpacingProps> = Template.bind({})
SemiPadding.args = { padding: 'semi' }
SemiPadding.storyName = `Semi padding (${semi}px)`

export const SemiPaddingTop : Story<ISpacingProps> = Template.bind({})
SemiPaddingTop.args = { paddingTop: 'semi' }
SemiPaddingTop.storyName = `Semi Top Padding (${semi}px)`

export const StandardPadding : Story<ISpacingProps> = Template.bind({})
StandardPadding.args = { padding: 'standard' }
StandardPadding.storyName = `Standard padding (${standard}px)`

export const SmallPadding : Story<ISpacingProps> = Template.bind({})
SmallPadding.args = { padding: 'small' }
SmallPadding.storyName = `Small padding (${small}px)`

export const TinyPadding : Story<ISpacingProps> = Template.bind({})
TinyPadding.args = { padding: 'tiny' }
TinyPadding.storyName = `Tiny padding (${tiny}px)`

export const MicroPadding : Story<ISpacingProps> = Template.bind({})
MicroPadding.args = { padding: 'micro' }
MicroPadding.storyName = `Micro padding (${micro}px)`

export const HugeTopPadding : Story<ISpacingProps> = Template.bind({})
HugeTopPadding.args = { paddingTop: 'huge' }
HugeTopPadding.storyName = `Huge top padding (${huge}px)`

export const LargeRightPadding : Story<ISpacingProps> = Template.bind({})
LargeRightPadding.args = { paddingRight: 'large' }
LargeRightPadding.storyName = `Large right padding (${large}px)`

export const SemiVerticalPadding : Story<ISpacingProps> = Template.bind({})
SemiVerticalPadding.args = { paddingY: 'semi' }
SemiVerticalPadding.storyName = `Semi vertical padding (${semi}px)`

export const SemiBottomPadding : Story<ISpacingProps> = Template.bind({})
SemiBottomPadding.args = { paddingBottom: 'semi' }
SemiBottomPadding.storyName = `Semi bottom padding (${semi}px)`

export const StandardLeftPadding : Story<ISpacingProps> = Template.bind({})
StandardLeftPadding.args = { paddingLeft: 'standard' }
StandardLeftPadding.storyName = `Standard left padding (${standard}px)`

export const SmallHorizontalPadding : Story<ISpacingProps> = Template.bind({})
SmallHorizontalPadding.args = { paddingX: 'small' }
SmallHorizontalPadding.storyName = `Small horizontal padding (${small}px)`

export const SmallTopPadding : Story<ISpacingProps> = Template.bind({})
SmallTopPadding.args = { paddingTop: 'small' }
SmallTopPadding.storyName = `Small top padding (${small}px)`

export const TinyRightPadding : Story<ISpacingProps> = Template.bind({})
TinyRightPadding.args = { paddingRight: 'small' }
TinyRightPadding.storyName = `Tiny right padding (${tiny}px)`

export const MicroBottomPadding : Story<ISpacingProps> = Template.bind({})
MicroBottomPadding.args = { paddingBottom: 'micro' }
MicroBottomPadding.storyName = `Micro bottom padding (${micro}px)`

export const NoPadding : Story<ISpacingProps> = Template.bind({})
NoPadding.args = { padding: 'none' }
NoPadding.storyName = `No padding (${none}px)`
