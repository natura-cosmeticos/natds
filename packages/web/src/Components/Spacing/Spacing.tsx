import * as React from 'react'

import clsx from 'clsx'
import { Box } from '@material-ui/core'
import { ISpacingProps } from './Spacing.props'
import { useStyles } from './Spacing.styles'

export const Spacing : React.FunctionComponent<ISpacingProps> = (
  props: ISpacingProps
) => {
  const {
    children,
    className,
    component: Component = 'div',
    display,
    // eslint-disable-next-line id-length
    m,
    margin,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    marginX,
    marginY,
    mb,
    ml,
    mr,
    mt,
    mx,
    my,
    // eslint-disable-next-line id-length
    p,
    padding,
    paddingBottom,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingX,
    paddingY,
    pb,
    pl,
    pr,
    pt,
    px,
    py,
    style,
    ...otherProps
  } = props

  const classes = useStyles({
    // eslint-disable-next-line id-length
    m,
    margin,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    marginX,
    marginY,
    mb,
    ml,
    mr,
    mt,
    mx,
    my,
    // eslint-disable-next-line id-length
    p,
    padding,
    paddingBottom,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingX,
    paddingY,
    pb,
    pl,
    pr,
    pt,
    px,
    py
  })

  return (
    <Box
      component={Component}
      className={clsx(classes.root, className)}
      display={display}
      {...otherProps}
      style={style}
    >
      {children}
    </Box>
  )
}

Spacing.displayName = 'Spacing'
export default Spacing
