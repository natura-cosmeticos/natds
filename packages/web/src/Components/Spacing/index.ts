import Spacing from './Spacing'

export { ISpacingProps } from './Spacing.props'
export { Spacing }
export default Spacing
