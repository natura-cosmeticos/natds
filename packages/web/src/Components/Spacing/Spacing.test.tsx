import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import Spacing from './Spacing'
import { spacingArgTypes } from './Spacing.argTypes'
import { Provider } from '../../Provider'

describe('Spacing component', () => {
  let testRenderer : TestRenderer.ReactTestRenderer = {} as TestRenderer.ReactTestRenderer

  describe('given no default theme is provided', () => {
    describe('when no props are provided', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Spacing />)

        expect(testRenderer).toMatchSnapshot()
      })
    })
  })

  describe('given a valid default theme is provided', () => {
    describe('when no props are provided', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Provider>
          <Spacing />
        </Provider>)

        expect(testRenderer).toMatchSnapshot()
      })
    })
    describe('when a `div` is provided as component', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Provider>
          <Spacing component="div" />
        </Provider>)

        expect(testRenderer).toMatchSnapshot()
      })
    })
    describe('when a `span` is provided as a component', () => {
      it('should render correctly', () => {
        testRenderer = TestRenderer.create(<Provider>
          <Spacing component="span" />
        </Provider>)

        expect(testRenderer).toMatchSnapshot()
      })
    })
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    describe.each(Object.keys(spacingArgTypes))('when a %p prop is provided', (prop) => {
      it('should render correctly', () => {
        const props = {
          [prop]: 'small'
        }

        testRenderer = TestRenderer.create(<Provider>
          <Spacing {...props} />
        </Provider>)
        expect(testRenderer).toMatchSnapshot()
      })
    })
  })
})
