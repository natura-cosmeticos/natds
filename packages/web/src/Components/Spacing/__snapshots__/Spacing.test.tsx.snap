// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing component given a valid default theme is provided when a "margin" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginBottom" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginLeft" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginRight" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginTop" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginX" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "marginY" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "padding" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingBottom" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingLeft" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingRight" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingTop" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingX" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a "paddingY" prop is provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a \`div\` is provided as component should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when a \`span\` is provided as a component should render correctly 1`] = `
<span
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given a valid default theme is provided when no props are provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;

exports[`Spacing component given no default theme is provided when no props are provided should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
/>
`;
